WordPress/gutenberg
# Allow setting a `defaultBlock` attribute on the navigation block

## What?
This adds a new `defaultBlock` attribute to allow site owners to customize the `core/navigation` block experience for end users.

## Why?
This solves for https://github.com/WordPress/gutenberg/issues/50982

## How?
This adds a new `defaultBlock` attribute and a check for it along with a quick sanity check if it fails it falls back to the `DEFAULT_BLOCK` constant.

## Testing Instructions
1. Open a post or page
2. Copy and paste this block markup into the editor:
```

```
3. Click + inside the navigation block.
4. Now category links should be the default when clicking + inside this navigation block.

### Testing Instructions for Keyboard
n/a
